### PR TITLE
NCE (sampler & loss) added.

### PR DESCRIPTION
**Pull request by leejayyoon**
NCE sampler & loss functions are added.

### CODE REVIEW

File: structured_prediction_baselines/modules/loss/nce_loss.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        ) 
+        # y_hat should not be normalized for BCEWithLogitLoss(),
+        # samples should be between [0,1] for each entry.
+        
+        p_n = torch.sum(p_n, dim=2)
+        # (batch, num_samples)
+        # y_hat.expand_as(samples): (batch, 1, labels) --> (batch, num_samples, labels)
+        # print("==============================================predicted_score.shape:{}".format(predicted_score.shape))
+        # print("==============================================p_n.shape:{}".format(p_n.shape))
+        new_score = predicted_score - p_n
```
Comment (dhruvdcoder) [inline structured_prediction_baselines/modules/loss/nce_loss.py:93]: pn is same as BCE loss or negative log-likelihood. Shouldn't the sign be the opposite?
